diesel/src/connection/statement_cache.rs

```diff
@@ -0,0 +1,54 @@
+use std::any::TypeId;
+use std::borrow::Cow;
+
+use backend::Backend;
+use query_builder::*;
+use result::QueryResult;
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+#[derive(Hash, PartialEq, Eq)]
+pub enum StatementCacheKey<DB: Backend> {
+    Type(TypeId),
+    Sql {
+        sql: String,
+        bind_types: Vec<DB::TypeMetadata>,
+    }
+}
```

Comment:
Yeah, I'm really not worried about the 40 byte overhead there. I do want to split the two caches into separate maps though since the `TypeId` can work with a phf which will reflect perf-wise